sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -70,10 +72,40 @@
     INNER JOIN sys.dm_exec_requests req
         ON c.connection_id = req.connection_id
     CROSS APPLY sys.dm_exec_sql_text(req.sql_handle) qt
+    CROSS APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) lqt
 WHERE sess.session_id != @@spid AND sess.status != 'sleeping'
 """,
 ).strip()
 
+# Query to fetch idle blocking sessions
+# with only basic sess and conn info
+# not joining with sys.dm_exec_requests
```

Comment:
The issue with not joining on `sys.dm_exec_requests` is that there is a lot of information lost about the blocker which is what users will try to investigate. It will also create inconsistencies in the data model we will have to support. For instance see all of the information contained here under the `db` fields and `sqlserver` fields:

<img width="892" alt="image" src="https://github.com/DataDog/integrations-core/assets/717862/137a90ff-f44a-4870-b61c-5715e43a9753">

Much of this will be lost with this approach.

I can't say for certain what the impact is of joining against `sys.dm_exec_requests` is for extreme outlier cases, but  so far I am not seeing the join present significantly higher plan complexity. It seems like the issue is incorporating more data at all. For instance see this plan on a fairly large database returned 1640 rows after all joins and filters are applied, which is the result of the join from 3410 rows in `SYSCONNECTIONS` and 3390 rows in `SYSREQUESTS`. The middle branch with leaf node is the requests table `SYSREQUESTS` (sys.dm_exec_requests). And you can see that a sort is applied automatically despite the query having no ORDER BY, likely because there are so many rows (I'm not sure why, perhaps @nenadnoveljic would know). Connections and requests are both large contributors to total cost, but remove either and you still have a fairly high cost plan. This one ran in 9 seconds!

<img width="1172" alt="image" src="https://github.com/DataDog/integrations-core/assets/717862/81852e48-051e-4773-bf3f-6920bb8a3e3e">

BUT if we apply a filter on all of these tables the second time we issue the query, I believe the cost will be *drastically* reduced. 

My proposal is that you can use the exact same query as above, but simply add a filter condition to each with the idle blocking session IDs identified in the first query. I believe this second query will have a very small number of IDs and will therefore run in a small fraction of the time. 

So rather than each branch of that plan operating on 3k+ rows, it will operate on ~10 rows or less after a Filter condition is added.

Reply:
This answer from ChatGPT could explain the sorting btw:

```
Yes, a merge join in SQL Server requires sorted data. A merge join is a type of join operation that combines two sorted input sets (usually the result of two separate queries) into a single, joined result set. It's an efficient join algorithm when the input data is already sorted on the join key because it allows for a linear comparison of the two sorted lists, resulting in better performance compared to nested loop or hash joins for larger datasets.

In your scenario, if the query involves a merge join and the primary key is being sorted, it's likely that the optimizer is using a merge join strategy to combine the data. The primary key sorting might be necessary to ensure that the data from both sides of the join is properly aligned and compared to perform the merge join operation effectively.

If the primary key is already sorted, the sorting operation might be part of the optimization process to ensure that the data is properly prepared for the merge join. Sorting the primary key might also be beneficial if the primary key order is different from the join key order or if the optimizer determines that a different execution plan with a sorted input results in better overall performance.

In summary, when you see a sort operation before a merge join in a query execution plan, it's generally because the merge join requires sorted data to work efficiently. The primary key sorting could be part of the optimization process to facilitate the merge join operation.
```

Reply:
It turns out `dm_exec_requests` dmv does not contain idle sessions. Inner joining `dm_exec_sessions` with `dm_exec_requests` will not return any idle blocking sessions. This prevent us reusing the same `ACTIVITY_QUERY` for regular activities and idle blocking sessions. The proposed change is to continue using a simplified activity query (not joining `dm_exec_requests` dmv ) for to fetch idle blocking sessions and get raw sql from `dm_exec_connections.most_recent_sql_handle`.

Reply:
Can you update the comment with this info?